runtime/src/accounts_db.rs

```diff
@@ -230,10 +232,10 @@ impl<'a> LoadedAccount<'a> {
         }
     }
 
-    pub fn loaded_hash(&self) -> &Hash {
+    pub fn loaded_hash(&self) -> Hash {
         match self {
-            LoadedAccount::Stored(stored_account_meta) => &stored_account_meta.hash,
-            LoadedAccount::Cached((_, cached_account)) => &cached_account.hash,
+            LoadedAccount::Stored(stored_account_meta) => *stored_account_meta.hash,
+            LoadedAccount::Cached((_, cached_account)) => cached_account.hash(),
```

Comment:
Ah clever, so if the hash isn't computed yet by the background service by this point, (for things like `bank.freeze()` that need the hash), it'll compute it here.

Reply:
Thanks. I am just implementing the required blocking, lazy behavior in a way that seems correct, consistent, performant. I think it fits Rust idioms sufficiently.